models/models.js

```diff
@@ -76,7 +84,7 @@ exports.fetchComments = (articleId) => {
     .then(({ rows }) => {
       const commentArray = rows;
       if (commentArray.length === 0) {
-        return Promise.reject({ status: 404, msg: "No article found" });
+        return [];
       }
       return commentArray;
     });
```

Comment:
You're along the right lines with multiple DB queries being needed, but to accomplish this we would typically be returning a Promise.all inside the controller on two functions, one function being the model and the other being to check if the article id exists. Promise.all rejects when _any_ of the promises inside it rejects.